nautobot_ssot/integrations/librenms/diffsync/adapters/librenms.py

```diff
@@ -67,41 +75,187 @@ def load_location(self, location: dict):
 
     def load_device(self, device: dict):
         """Load Device objects from LibreNMS into DiffSync models."""
+        # Ensure device is a dictionary
+        if not isinstance(device, dict):
+            self.job.logger.warning(f"Device data is not a dictionary: {type(device)} - {device}")
+            self.failed_import_devices.append(device)
+            return
+
+        # Get the hostname field to use
+        hostname_field = (
+            os.getenv("NAUTOBOT_SSOT_LIBRENMS_HOSTNAME_FIELD", "sysName")
+            if self.job.hostname_field == "env_var"
+            else self.job.hostname_field or "sysName"
+        )
+
         if self.job.debug:
-            self.job.logger.debug(f'Loading LibreNMS Device {device["sysName"]}')
+            self.job.logger.debug(f"LibreNMS Adapter Loading LibreNMS Device {device[hostname_field]}")
 
         if device["os"] != "ping":
```

Comment:
You might want to rewrite this to instead throw the logging error and continue. It'd be clearer as this block of text is quite large and hard to determine what goes with what.